homeWork3/lesson1/lesson1/Infrastructure/Commands/LambdaCommand.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace lesson1.Infrastructure.Commands.Base;
+
+public class LambdaCommand : CreateCommand
+{
+    public LambdaCommand(Action<object?> OnExecute, Func<object?, bool>? OnCanExecute = null)
+    {
+        _onExecute = OnExecute;
+        _onCanExecute = OnCanExecute;
+    }
+
+    private readonly Action<object?> _onExecute;
+    private readonly Func<object?, bool>? _onCanExecute;
+
+    public override void Execute(object? parameter)
+    {
+        _onExecute(parameter);

```

Comment:
```suggestion
        if(_onCanExecute?.Invoke(parameter) ?? true)
             _onExecute(parameter);
```
Стоит дополнительно выполнить проверку.